Packs/QuestKace/Integrations/QuestKace/QuestKace.py

```diff
@@ -0,0 +1,907 @@
+        response = client.delete_ticket_request(ticket_id)
+    except Exception as e:
+        raise DemistoException(e)
+    if response.get('Result') == 'Success':
+        context = {}
+        old_context = demisto.dt(demisto.context(), f'QuestKace.Ticket(val.ID === {ticket_id})')
+        if old_context:
+            if isinstance(old_context, list):
+                old_context = old_context[0]
+            old_context['IsDeleted'] = True
+            context = {
+                f'QuestKace.Ticket(val.ID === obj.ID)': old_context
+            }
+        return f'Ticket was deleted successfully. Ticket number {ticket_id}', context, {}
+    else:
+        raise DemistoException(f'Error while deleting the ticket.')
+
+
+def fetch_incidents(client: Client, fetch_time: str, fetch_shaping: str, last_run: Dict, fetch_limit: str,
+                    fetch_queue_id: Optional[list] = None, fetch_filter: Optional[str] = None) -> list:
+    """
+    This function will execute each interval (default is 1 minute).
+    Args:
+        client (Client): Quest Kace Client
+        fetch_time: time interval for fetch incidents.
+        fetch_shaping: shaping for the request.
+        fetch_filter: custom filters for the request.
+        fetch_limit: limit for number of fetch incidents per fetch.
+        fetch_queue_id: queue id for fetch, if not given then fetch runs on all tickets in the system
+        last_run (dateparser.time): The greatest incident created_time we fetched from last fetch
+    Returns:
+        incidents: Incidents that will be created in Demisto
+    """
+    if not fetch_queue_id or fetch_queue_id[0] == 'All':
+        fetch_queue_id = get_queue_ids(client)
+    time_format = '%Y-%m-%dT%H:%M:%SZ'
+    if not last_run:  # if first time running
+        new_last_run = {'last_fetch': parse_date_range(fetch_time, date_format=time_format)[0]}
+    else:
+        new_last_run = last_run
+
+    if not fetch_shaping:
+        fetch_shaping = shaping_fetch(client, fetch_queue_id)
+
+    parsed_last_time = datetime.strptime(new_last_run.get('last_fetch', ''), time_format)
+    filter_after_last_run = f'created gt {parsed_last_time}'
+    fetch_filter_for_query = f'{filter_after_last_run}'
+    if fetch_queue_id:
+        queue_id_str = ';'.join(fetch_queue_id)
+        filter_by_queue_id = f'hd_queue_id in {queue_id_str}'
+        fetch_filter_for_query = f'{fetch_filter_for_query},{filter_by_queue_id}'
+    if fetch_filter:
+        fetch_filter_for_query = f'{fetch_filter_for_query},{fetch_filter}'
+    demisto.info(f"Fetching Incident has Started,\n"
+                 f"Fetch filter is {fetch_filter_for_query}\n"
+                 f"Last fetch was on {str(parsed_last_time)}")
+    client.update_token()
+    items: dict = client.tickets_list_request(fetch_shaping, fetch_filter_for_query)
+    items: list = items.get('Tickets', [])
+    incidents, parsed_last_time = parse_incidents(items, fetch_limit, time_format, parsed_last_time)
```

Comment:
have a different name, or else it is misleading to think that this vars have the same value.
```suggestion
    incidents, last_incident_time = parse_incidents(items, fetch_limit, time_format, parsed_last_time)
```